### PR TITLE
dropdown

### PR DESCRIPTION
**Pull request by Amitrawat21**
dropdown component

### CODE REVIEW

**Review by emailpocketxtra — COMMENTED** [review]
What is the purpose of 2 interfaces, DropDownInterface.ts / DropDownInterfaceTwo.ts??

File: src/components/dropdown/DropDown.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+
+import React, { useState } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import { responsiveHeight, responsiveWidth } from 'react-native-responsive-dimensions';
+import { DropDownProps } from '../../interface/DropDown/DropDownInterfaceTwo';
+
+ export const DropDown: React.FC<DropDownProps> = ({
```
Comment (rohan-arya-w) [inline src/components/dropdown/DropDown.tsx]: Changed the const name

File: src/components/dropdown/DropDown.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    height = 7,
+    width = 65,
```
Comment (rohan-arya-w) [inline src/components/dropdown/DropDown.tsx]: remove the hardcoded width

File: src/components/dropdown/DropDown.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            width: responsiveWidth(width),
+            borderRadius: responsiveHeight(borderRadius),
+          }}
+          activeOpacity={0.8}
+          onPress={toggleOptions}>
+          <Text style={{ color: '#E4E4E4' }}>{selectedItem || selectedLabel}</Text>
+          <Icon
+            style={{
+              transform: [{ rotate: showOptions ? '180deg' : '0deg' }],
+              color: '#808080',
```
Comment (rohan-arya-w) [inline src/components/dropdown/DropDown.tsx]: remove the hardcoded color

File: src/components/dropdown/DropDown.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
+
+import React, { useState } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import { responsiveHeight, responsiveWidth } from 'react-native-responsive-dimensions';
+import { DropDownProps } from '../../interface/DropDown/DropDownInterfaceTwo';
+
+export const CustomDropDownComponent: React.FC<DropDownProps> = ({
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx]: change the interface name

File: src/components/dropdown/DropDown.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  height,
+  width,
+  backgroundColor,
+  borderRadius,
+  Coordinators,
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx]: what are coordinators?? give proper prop names

File: src/components/dropdown/DropDown.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  selectedLabel,
+  margin,
+  textColor,
+  iconColor,
+  iconsize
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx]: use camelcase

File: src/components/dropdown/DropDown.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        style={{
+          backgroundColor : backgroundColor?backgroundColor :"black",
+          justifyContent : 'space-between',
+          flexDirection : "row",
+          alignItems :'center',
+          padding: responsiveHeight(1),
+          height: responsiveHeight(height?height:10),
+          width: responsiveWidth(width?width:80),
+          borderRadius: responsiveHeight(borderRadius?borderRadius : 1),
+
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx:47]: remove extra lines

File: src/components/dropdown/DropDown.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        />
+      </TouchableOpacity>
+      {showOptions && (
+        <View style={{ position: 'absolute', top: '100%'}}>
+          {Coordinators?.map((coordinator, i) => (
+            <TouchableOpacity
+              style={{
+                backgroundColor :backgroundColor?backgroundColor: "black",
+                justifyContent : 'space-between',
+                flexDirection : "row",
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx:69]: use single or double quotes

File: src/components/dropdown/DropDown.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                width: responsiveWidth(width?width:80),
+                borderRadius: responsiveHeight(borderRadius?borderRadius:1),
+                margin : responsiveWidth(margin?margin:0.1)            
+                
+      
+              }}
+              key={i}
+              onPress={() => onSelectItem(coordinator.name)}>
+              <Text key={String(i)} style={{ color: textColor?textColor:'#E4E4E4' }}>
+                {coordinator.name}
```
Comment (tabassum2507) [inline src/components/dropdown/DropDown.tsx]: remove coordinator and use other name

File: src/components/footer/index.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
 export { default as ContainedButton } from '../buttons/containedButton';
-
+import { DropDownProps } from '../../interface/DropDown/DropDownInterfaceTwo';
```
Comment (tabassum2507) [inline src/components/footer/index.ts]: why are you making changes in footer?

File: src/interface/DropDown/DropDownInterfaceTwo.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+ interface DropdownOption {
```
Comment (tabassum2507) [inline src/interface/DropDown/DropDownInterfaceTwo.ts]: use camelcase

**Review by tabassum2507 — CHANGES_REQUESTED** [review]

**Review by rohan-arya-w — APPROVED** [review]

**Review by rohan-arya-w — APPROVED** [review]

**Review by tabassum2507 — APPROVED** [review]

File: src/components/dropdown/DropDown.tsx (hunk 8 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx:69]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 7 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx:47]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 3 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 2 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 1 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

File: src/interface/DropDown/DropDownInterfaceTwo.ts (hunk 11 above)
Comment (Amitrawat21) [reply src/interface/DropDown/DropDownInterfaceTwo.ts]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 9 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 6 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

File: src/components/dropdown/DropDown.tsx (hunk 4 above)
Comment (Amitrawat21) [reply src/components/dropdown/DropDown.tsx]: resolved

**Pull request merged**